tests/ha/check_hae_active.py

```diff
@@ -0,0 +1,85 @@
+# SUSE's openQA tests
+#
+# Copyright 2025 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: Check that SUT has the HA Extension Enabled. If it's not, soft fail and work around
+# Maintainer: qe-sap@suse.com
+
+from testapi import *
+import json
+
+def check_suseconnect(self):
+    '''
+    Check that SUT has suseconnect-ng installed. Install it if it's not there
+    '''
+    no_suseconnect = int(script_run("which SUSEConnect"))
+    if (no_suseconnect):
+        record_soft_failure("bsc#1238913 - SUSEConnect is not installed by default")
+        perl.utils.zypper_call("in suseconnect-ng")
+
+def check_product_registration(self, status=""):
+    '''
+    Check SUT is correctly registered to the SCC proxy. Re-register if it's not registered to
+    the proxy, if it's not registered or if it's missing /etc/SUSEConnect
+    '''
+    no_suseconnect = int(script_run("test -f /etc/SUSEConnect"))
+    scc_proxy_reg = False
+    build = get_required_var("BUILD")
+    scc_url = get_var("SCC_URL", f"http://all-{build}.proxy.scc.suse.de")
+
+    if (not no_suseconnect):
```

Comment:
this `not` `no_` double negation is a little bit confusing

Reply:
I know. But `no_suseconnect` is later used without the negation, so figured it was OK. Alternative is doing:
```
is_suseconnect = not int(script_run("test -f /etc/SUSEConnect"))
```
In line 26, and then `if (is_suseconnect)` here and later in line 38:
```
if (not is_suseconnect or scc_status[0]["status"] == "Not Registered" or not scc_proxy_reg):
```
Would that be better?

Reply:
```
etc_suseconnect = ( 0 == int(script_run("test -f /etc/SUSEConnect")))
```

Reply:
Changed it to `etc_suseconnect_exists`